external/webserver/http_client.c

```diff
@@ -167,6 +167,49 @@ int http_client_release(struct http_client_t *client)
 	return HTTP_OK;
 }
 
+static int process_trailer_header(char *buf, unsigned int buf_len, struct http_req_message *req,
+				struct http_keyvalue_list_t *params,
+				struct http_client_t *client, unsigned int sentence_start)
+{
+	char *header_field_value = NULL;
+	unsigned int sentence_end;
+	char *entity;
+	unsigned int trailter_len;
+
+	/* Process the Trailer */
+	header_field_value = http_keyvalue_list_find(params, "Trailer");
+	if (strncmp(header_field_value, "(null)", 7) == 0) {
+		return -1;
+	}
+
+	/* match the Trailer field */
+	if (strncmp(buf + sentence_start, header_field_value, strlen(header_field_value))
+		 || *(buf + sentence_start + strlen(header_field_value)) != ':' ) {
+		return -1;
+	}
+
+	HTTP_LOGD("Trailer header: %s\n", header_field_value);
+	sentence_end = http_find_first_crlf(buf, buf_len, sentence_start);
+	if (sentence_end < 0 || sentence_end - sentence_start == 0) {
+		HTTP_LOGE("Error: Not accord with chunked encoding\n");
+		return -1;
+	}
+
+	/* copy Trailer data */
+	trailter_len = sentence_end - sentence_start;
+	entity = zalloc(trailter_len + 1);
```

Comment:
I think this causes build warning for type mismatch.
```
entity = (char *)zalloc(....);
```

Reply:
done